.travis.yml

```diff
@@ -184,6 +184,30 @@ jobs:
       after_failure:
         - tests/run-tests --print-failures
 
+    # unix with 32bit float
+    - stage: test
+      env: NAME="unix port build and tests with MICROPY_FLOAT_IMPL_FLOAT"
+      script:
+        - make ${MAKEOPTS} -C mpy-cross
+        - make ${MAKEOPTS} -C ports/unix submodules
+        - make ${MAKEOPTS} -C ports/unix CFLAGS_EXTRA="-DMICROPY_FLOAT_IMPL=MICROPY_FLOAT_IMPL_FLOAT"
+        - make ${MAKEOPTS} -C ports/unix test
+      after_failure:
+        - tests/run-tests --print-failures
+
+    # unix with clang and 32bit float
+    - stage: test
+      env: NAME="unix port build and tests with MICROPY_FLOAT_IMPL_FLOAT and clang"
+      install:
+        - sudo apt-get install clang
+      script:
+        - make ${MAKEOPTS} -C mpy-cross CC=clang
+        - make ${MAKEOPTS} -C ports/unix submodules
+        - make ${MAKEOPTS} -C ports/unix CC=clang CFLAGS_EXTRA="-DMICROPY_STACKLESS=1 -DMICROPY_STACKLESS_STRICT=1 -DMICROPY_FLOAT_IMPL=MICROPY_FLOAT_IMPL_FLOAT"
+        - make ${MAKEOPTS} -C ports/unix CC=clang test
+      after_failure:
+        - tests/run-tests --print-failures
+
```

Comment:
The number of jobs are growing, which means slower CI builds.  Perhaps these two can be combined into one job?  Would need a "make clean" in between building with gcc and clang.

Also, the clang config does not need the stackless configs set, that was just to test stackless and here should not be set.

Reply:
I thought of that already at one point. Combine just these 2, or also the unix stackless one since that also uese clang?

Reply:
How about putting the clang 32-bit float into the existing unix-clang job, and the gcc 32-bit float into unix nan-box job?

Reply:
Sure.

Btw the number of jobs isn't yet the limiting factor in Travis, it's the OSX build. By the time it effectively starts all other jobs usually finished already..

Reply:
If that's the case the osx job should go at the top, to start first.